vnext/Microsoft.ReactNative/Fabric/platform/react/renderer/graphics/Color.cpp

```diff
@@ -7,58 +7,113 @@
+      {"TextFillColorDisabled", { 0x5C, 0x00, 0x00, 0x00 }},
+      // Arguably only the control-agnostic platform colors should be respected, and 
+      // Button should be updated to use those instead, assuming that still holds up
+      // in high contrast and such.
+      {"ButtonBackgroundPressed", { 0x4D, 0xF9, 0xF9, 0xF9 }}, // ControlFillColorTertiary
+      {"ButtonForegroundPressed", { 0x9E, 0x00, 0x00, 0x00 }}, // TextFillColorSecondary
+      {"ButtonForegroundPointerOver", { 0xE4, 0x00, 0x00, 0x00 }}, // TextFillColorPrimary
+      {"ButtonBackground", { 0xB3, 0xFF, 0xFF, 0xFF }}, // ControlFillColorDefault
+      {"ButtonBorderBrush", { 0x29, 0x00, 0x00, 0x00 }}, // gradient from ControlStrokeColorSecondary to ControlStrokeColorDefault
+      {"ButtonForeground", { 0xE4, 0x00, 0x00, 0x00 }}, // TextFillColorPrimary
+      {"ButtonBackgroundDisabled", { 0x4D, 0xF9, 0xF9, 0xF9 }}, // ControlFillColorDisabled
+      {"ButtonBorderBrushDisabled", { 0x0F, 0x00, 0x00, 0x00 }}, // ControlStrokeColorDefault
+      {"ButtonForegroundDisabled", { 0x5C, 0x00, 0x00, 0x00 }}, // TextFillColorDisabled
+      {"ButtonBackgroundPointerOver", { 0x80, 0xF9, 0xF9, 0xF9 }}, // ControlFillColorSecondary
+      {"ButtonBorderBrushPointerOver", { 0x66, 0x00, 0x00, 0x00 }}, // gradient from ControlStrokeColorOnAccentSecondary to ControlStrokeColorOnAccentDefault
+      {"ButtonBorderBrushPressed", { 0x00, 0xFF, 0xFF, 0xFF }}, // ControlFillColorTransparent
+    };
+
+    if (!color->m_platformColor.empty()) {
+      auto result = s_windowsColors.find(color->m_platformColor);
+      if (result != s_windowsColors.end()) {
+        return result->second;
+      }
+    }
+
+    return color->m_color;
+  }
+
+  D2D1::ColorF SharedColor::AsD2DColor() const {
+    winrt::Windows::UI::Color color = ResolvePlatformColor(m_color.get());
+    return {
+        color.R / 255.0f,
+        color.G / 255.0f,
+        color.B / 255.0f,
+        color.A / 255.0f };
   }
 
-  return colorComponentsFromColor(color).alpha > 0;
-}
+  winrt::Windows::UI::Color SharedColor::AsWindowsColor() const {
+    return ResolvePlatformColor(m_color.get());
+  }
 
 #ifndef CORE_ABI
-xaml::Media::Brush SharedColor::AsWindowsBrush() const {
-  if (!m_color)
-    return nullptr;
-  if (!m_color->m_platformColor.empty()) {
-    return Microsoft::ReactNative::BrushFromColorObject(winrt::to_hstring(m_color->m_platformColor));
+  xaml::Media::Brush SharedColor::AsWindowsBrush() const {
+    if (!m_color)
+      return nullptr;
+    if (!m_color->m_platformColor.empty()) {
+      return Microsoft::ReactNative::BrushFromColorObject(winrt::to_hstring(m_color->m_platformColor));
+    }
+    return xaml::Media::SolidColorBrush(m_color->m_color);
   }
-  return xaml::Media::SolidColorBrush(m_color->m_color);
-}
 #endif // CORE_ABI
 
-SharedColor colorFromComponents(ColorComponents components) {
-  float ratio = 255;
+  SharedColor colorFromComponents(ColorComponents components) {
```

Comment:
Why the sudden indentation? Do you need to re-run `yarn format`?

Reply:
Ugh, thought I had removed those, must have re-added in my last edit. VS/VSCode's checked in config for the repo has "format on save" turned on, and the files are out of sync with the checked in clang format file. So it seems like either the format rules picked by the editor needs to be updated or there needs to be a whitespace-only pass on all the checked in Fabric files. I was explicitly trying to avoid getting into that mess for this change, but one of the commits when awry.

Reply:
FWIW `yarn format` isn't running for me:

> \react-native-windows> yarn format
yarn run v1.22.19
$ format-files -i -style=file -assume-filename=../.clang-format
'format-files' is not recognized as an internal or external command,
operable program or batch file.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Reply:
Rebased to fix the commit that messed with whitespace.